pkg/broker/api.go

```diff
@@ -436,37 +436,34 @@ func (b *AwsBroker) Update(request *osb.UpdateInstanceRequest, c *broker.Request
 	// Get the service instance
 	instance, err := b.db.DataStorePort.GetServiceInstance(request.InstanceID)
 	if err != nil {
-		desc := fmt.Sprintf("Failed to get the service instance %s: %v", request.InstanceID, err)
+		desc := fmt.Sprintf("Failed to get the service instance %q: %v", request.InstanceID, err)
 		return nil, newHTTPStatusCodeError(http.StatusInternalServerError, "", desc)
 	} else if instance == nil {
-		desc := fmt.Sprintf("The service instance %s was not found.", request.InstanceID)
+		desc := fmt.Sprintf("The service instance %q was not found.", request.InstanceID)
 		return nil, newHTTPStatusCodeError(http.StatusBadRequest, "", desc)
 	}
 
 	// Verify that we're not changing the plan (this should never happen since
 	// we're setting `plan_updateable: false`, but better safe than sorry)
 	if request.PlanID != nil && *request.PlanID != instance.PlanID {
-		desc := fmt.Sprintf("The service plan cannot be changed from %s to %s.", instance.PlanID, *request.PlanID)
+		desc := fmt.Sprintf("The service plan cannot be changed from %q to %q.", instance.PlanID, *request.PlanID)
 		return nil, newHTTPStatusCodeError(http.StatusBadRequest, "", desc)
 	}
 
 	// Get the service
 	service, err := b.db.DataStorePort.GetServiceDefinition(request.ServiceID)
 	if err != nil {
-		desc := fmt.Sprintf("Failed to get the service %s: %v", request.ServiceID, err)
+		desc := fmt.Sprintf("Failed to get the service %q: %v", request.ServiceID, err)
 		return nil, newHTTPStatusCodeError(http.StatusInternalServerError, "", desc)
 	} else if service == nil {
-		desc := fmt.Sprintf("The service %s was not found.", request.ServiceID)
+		desc := fmt.Sprintf("The service %q was not found.", request.ServiceID)
 		return nil, newHTTPStatusCodeError(http.StatusBadRequest, "", desc)
 	}
 
-	// Get the plan and verify that it has updatable parameters
+	// Get the plan
 	plan := getPlan(service, instance.PlanID)
 	if plan == nil {
-		desc := fmt.Sprintf("The service plan %s was not found.", instance.PlanID)
-		return nil, newHTTPStatusCodeError(http.StatusBadRequest, "", desc)
-	} else if plan.Schemas.ServiceInstance.Update == nil {
```

Comment:
This is the only major change in this PR. Instead of failing early if the plan has no updatable parameters, we'll let the code below check whether any parameter values are being changed. If the parameter values are the same (i.e. the update is restoring the original configuration), we'll return successfully with no CFN stack update required.